gemini/use-cases/retrieval-augmented-generation/utils/intro_multimodal_rag_utils.py

```diff
@@ -56,71 +56,41 @@ def get_text_embedding_from_text_embedding_model(
-        embedding_size: The size of the embedding vector. Defaults to 1408.
+        image_uri (str): The URI (Uniform Resource Identifier) of the image to process.
+        text (Optional[str]): Optional contextual text to guide the embedding generation. Defaults to "".
+        embedding_size (int): The desired dimensionality of the output embedding. Defaults to 512.
+        return_array (Optional[bool]): If True, returns the embedding as a NumPy array.
+        Otherwise, returns a list. Defaults to False.
 
     Returns:
-        A list representing the image embedding.
-
-    Important Note:
-    - Supported dimensions for Image embeddings: [128, 256, 512, 1408]
-    - Larger embedding sizes can capture finer details and patterns in images, leading to more accurate image classification and object detection.
-    - However, larger embedding sizes also increase latency.
-    - There is a trade-off between embedding size, quality, and latency.
-    - The choice of embedding size should be carefully considered based on the specific use case, available computational resources, and desired performance level.
+        list: A list containing the image embedding values. If `return_array` is True, returns a NumPy array instead.
     """
-
-    # Create a client to interact with the Vertex AI Prediction Service
-    client = aiplatform.gapic.PredictionServiceClient(
-        client_options={"api_endpoint": "us-central1-aiplatform.googleapis.com"}
-    )
-
-    # Specify the endpoint of the deployed multimodal embedding model
-    endpoint_multimodalembedding = f"projects/{project_id}/locations/us-central1/publishers/google/models/multimodalembedding@001"
-
-    # Prepare the input instance for the embedding model
-    instance = struct_pb2.Struct()
+    # image = Image.load_from_file(image_uri)
+    image = vision_model_Image.load_from_file(image_uri)
     if text:
-        instance["text"] = text
-
-    # Convert the image content to Base64 encoded string if provided
-    if image_uri:
-        image_content = load_image_bytes(image_uri)
-        encoded_content = b64encode(image_content).decode("utf-8")
-        instance["image"] = {"bytesBase64Encoded": encoded_content}
-
-    instances = [instance]
-
-    # Set the embedding size parameter
-    parameters = {"dimension": embedding_size}
-
-    # Send the prediction request to the embedding model
-    response = client.predict(
-        endpoint=endpoint_multimodalembedding,
-        instances=instances,
-        parameters=parameters,
-    )
-    image_embedding = [v for v in response.predictions[0].get("imageEmbedding", [])]
+        embeddings = multimodal_embedding_model.get_embeddings(
+            image=image, contextual_text=text, dimension=embedding_size
+        )  # 128, 256, 512, 1408
+    else:
+        embeddings = multimodal_embedding_model.get_embeddings(
+            image=image, dimension=embedding_size
+        )  # 128, 256, 512, 1408
```

Comment:
You actually don't need the if/else here. The SDK handles it.
```suggestion
    embeddings = multimodal_embedding_model.get_embeddings(
        image=image, contextual_text=text, dimension=embedding_size
    )  # 128, 256, 512, 1408
```

Reply:
Fixed!